libultraship/libultraship/Archive.cpp

```diff
@@ -28,7 +28,7 @@ namespace Ship {
 
 	std::shared_ptr<Archive> Archive::CreateArchive(const std::string& archivePath, int fileCapacity)
 	{
-		Archive* archive = new Archive(archivePath, true);
```

Comment:
Was this just a memory leak before? I have a vague notion of what `std::make_shared` does, but did it previously free the allocated Archive? Likewise, does the way you instantiate `archive` actually create the Archive? (I assume it does, but just trying to look for potential issues).

Reply:
Yes, this is it fix a memory leak.
As a side note, as far as I know, new memory leak were introduced recently and I also need to fix those.